functions/src/index.ts

```diff
@@ -219,7 +219,7 @@ export const executeFullIndexOperation = functions.tasks
     const newSuccessCount = pastSuccessCount + records.length;
     const newErrorCount = pastErrorCount;
 
-    if (snapshot.size === DOCS_PER_INDEXING) {
+    if (snapshot.size >= DOCS_PER_INDEXING) {
```

Comment:
There will be no impact with this change since the limit is set to 250 so the size will never be greater than 250.

Reply:
I captured this log where the indexing job receiving 266 documents from Firestore and consequently terminated the indexing job (even when the limit was set to 250).

```
  {
    "textPayload": "[ 'records.length', 266 ]",
    "insertId": "65702579000b34f8d955faad",
    "resource": {
      "type": "cloud_function",
      "labels": {
        "project_id": "monster-lab-2a47c",
        "region": "us-east1",
        "function_name": "ext-firestore-algolia-search-executeFullIndexOperation"
      }
    },
```

Full gist https://gist.github.com/LucasSwitz/e6ac3682e0036e53e7e1c4bc6841bdda.